Toolbar Improvements: Active is programmable

@brendancol just hit the example where he wanted to hide the toolbar (`toolbar_location=None`) but wanted zooming on.

We recently just turned zoomwheel (#3003) off by default, but the only way to turn it on is by clicking the toolbar icon.

We could add an argument to WheelZoomTool `active=True` for example, but it seems like we'd want something that worked across all tools and was consistent.

The challenge is that you should only have one tool active in a group at any one time. For WheelZoom this doesn't matter. It lives in the `scroll` group and there's only one tool in the `scroll` group, but for `pan` events in particular it's a problem.

I'm not entirely sure what a clean / obvious spelling is. We could have `active=True` on the invidivual models assuming that if you're using the Model interface you should know what you're doing and then have `figure` and `chart` have something like active_tools='list,of,tools' and it makes checks on it?


Worth noting that you can programmatically click the WheelZoomTool with if your toolbar is visible (but not if you've set `toolbar_location=None`)
````html
<script>
    Bokeh.$('.bk-button-bar-list[type="scroll"] button').click()
</script>
````

Update: if you have set `toolbar_location=None` you can reach into the guts of Bokeh to turn your WheelZoom on as follows:

````js
for (plot_index in Bokeh.index) {
 var plot_tools = Bokeh.index[plot_index].tools;
 for (tool_index in plot_tools) { 
   var tool = plot_tools[tool_index];
   if (tool.mget('tool_name') == 'Wheel Zoom') {
     tool.mset('active', true);
   }
 }
}
````

Hi @bryevdv, I'd love to take a look, unfortunately I'm having some issues installing bokeh from github. I've not used the developer version of bokeh before, so am following [these instruction](http://bokeh.pydata.org/en/latest/docs/dev_guide/setup.html#devguide-setup). I am using Ubuntu 14.04.4. I follow as far as cd'ing into `bokehjs`, however when I run `npm install`, I get the following:
`
npm ERR! not a package /home/btrav/tmp/npm-12834-0DIVdNSF/1466144537576-0.13068444072268903/tmp.tgz
npm ERR! Error: ENOENT, open '/home/btrav/tmp/npm-12834-0DIVdNSF/1466144537576-0.13068444072268903/package/package.json'
npm ERR! If you need help, you may report this log at:
npm ERR!     <http://github.com/isaacs/npm/issues>
npm ERR! or email it to:
npm ERR!     <npm-@googlegroups.com>

npm ERR! System Linux 3.13.0-88-generic
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! cwd /home/btrav/Documents/bokeh/bokehjs
npm ERR! node -v v0.10.25
npm ERR! npm -v 1.3.10
npm ERR! path /home/btrav/tmp/npm-12834-0DIVdNSF/1466144537576-0.13068444072268903/package/package.json
npm ERR! code ENOENT
npm ERR! errno 34
`

--- a/bokeh/charts/chart.py
+++ b/bokeh/charts/chart.py
@@ -30,7 +30,7 @@ from ..models import (
     Plot, HoverTool, FactorRange
 )
 from ..plotting import DEFAULT_TOOLS
-from ..plotting.helpers import _process_tools_arg, _glyph_function
+from ..plotting.helpers import _process_tools_arg, _glyph_function, _process_active_tools
 from ..core.properties import Auto, Either, Enum, String, Override
 from ..util.deprecate import deprecated
 from ..util._plot_arg_helpers import _convert_responsive
@@ -128,6 +128,10 @@ class Chart(Plot):
             kwargs['sizing_mode'] = _convert_responsive(kwargs['responsive'])
             del kwargs['responsive']
 
+        self._active_drag = kwargs.pop('active_drag', 'auto')
+        self._active_scroll = kwargs.pop('active_scroll', 'auto')
+        self._active_tap = kwargs.pop('active_tap', 'auto')
+
         title_text = kwargs.pop("title", None)
 
         super(Chart, self).__init__(*args, **kwargs)
@@ -156,7 +160,7 @@ class Chart(Plot):
         self._tooltips = []
 
         if hasattr(self, '_tools'):
-            self.create_tools(self._tools)
+            self.create_tools(self._tools, self._active_drag, self._active_scroll, self._active_tap)
 
     def add_renderers(self, builder, renderers):
         self.renderers += renderers
@@ -194,7 +198,7 @@ class Chart(Plot):
         if ygrid:
             self.make_grid(1, self._yaxis.ticker)
 
-    def create_tools(self, tools):
+    def create_tools(self, tools, active_drag, active_scroll, active_tap):
         """Create tools if given tools=True input.
 
         Only adds tools if given boolean and does not already have
@@ -209,8 +213,9 @@ class Chart(Plot):
 
         if len(self.toolbar.tools) == 0:
             # if no tools customization let's create the default tools
-            tool_objs = _process_tools_arg(self, tools)
+            tool_objs, tool_map = _process_tools_arg(self, tools)
             self.add_tools(*tool_objs)
+            _process_active_tools(self.toolbar, tool_map, self._active_drag, self._active_scroll, self._active_tap)
 
     def start_plot(self):
         """Add the axis, grids and tools
@@ -219,7 +224,7 @@ class Chart(Plot):
         self.create_grids(self._xgrid, self._ygrid)
 
         if self.toolbar.tools:
-            self.create_tools(self._tools)
+            self.create_tools(self._tools, self._active_drag, self._active_scroll, self._active_tap)
 
         if len(self._tooltips) > 0:
             self.add_tools(HoverTool(tooltips=self._tooltips))

--- a/bokeh/core/compat/bokeh_renderer.py
+++ b/bokeh/core/compat/bokeh_renderer.py
@@ -59,7 +59,7 @@ class BokehRenderer(Renderer):
     def close_figure(self, fig):
         "Complete the plot: add tools."
         # Add tools
-        tool_objs = _process_tools_arg(self.plot, DEFAULT_TOOLS)
+        tool_objs, tools_map = _process_tools_arg(self.plot, DEFAULT_TOOLS)
         self.plot.add_tools(*tool_objs)
 
         # Simple or Grid plot setup

--- a/bokeh/core/compat/bokeh_renderer.py
+++ b/bokeh/core/compat/bokeh_renderer.py
@@ -84,7 +84,7 @@ class BokehRenderer(Renderer):
 
                 _plot.title = ""
                 # and add new tools
-                _tool_objs = _process_tools_arg(_plot, DEFAULT_TOOLS)
+                _tool_objs, _tool_map = _process_tools_arg(_plot, DEFAULT_TOOLS)
                 _plot.add_tools(*_tool_objs)
                 # clean the plot ref from axis and grids
                 _plot_rends = subrends[i]

--- a/bokeh/models/tests/test_tools.py
+++ b/bokeh/models/tests/test_tools.py
@@ -1,0 +1,12 @@
+from __future__ import  absolute_import
+
+import bokeh.models.tools as tools
+
+def test_Toolbar():
+    tb = tools.Toolbar()
+    assert tb.active_drag == 'auto'
+    assert tb.active_scroll == 'auto'
+    assert tb.active_tap == 'auto'
+
+    # TODO (bev) validate entire list of props
+

--- a/bokeh/models/tests/test_tools.py
+++ b/bokeh/models/tests/test_tools.py
@@ -9,4 +9,3 @@ def test_Toolbar():
     assert tb.active_tap == 'auto'
 
     # TODO (bev) validate entire list of props
-

--- a/bokeh/models/tools.py
+++ b/bokeh/models/tools.py
@@ -25,7 +25,7 @@ from __future__ import absolute_import
 from ..model import Model
 from ..core.properties import abstract, Float, Color
 from ..core.properties import (
-    Any, Bool, String, Enum, Instance, Either, List, Dict, Tuple, Override
+    Any, Auto, Bool, String, Enum, Instance, Either, List, Dict, Tuple, Override
 )
 from ..core.enums import Dimension, Location
 
@@ -54,6 +54,25 @@ class Tool(Model):
     The Plot that this tool will act on.
     """)
 
+@abstract
+class Action(Tool):
+    pass
+
+@abstract
+class Drag(Tool):
+    pass
+
+@abstract
+class Scroll(Tool):
+    pass
+
+@abstract
+class Tap(Tool):
+    pass
+
+@abstract
+class Inspection(Tool):
+    pass
 
 @abstract
 class ToolbarBase(LayoutDOM):
@@ -76,12 +95,22 @@ class ToolbarBase(LayoutDOM):
     # orientation.
     sizing_mode = Override(default=None)
 
-
 class Toolbar(ToolbarBase):
     """ Hold tools to display for a single plot.
 
     """
 
+    active_drag = Either(Auto, Instance(Drag), help="""
+    Specify a drag tool to be active when the plot is displayed.
+    """)
+
+    active_scroll = Either(Auto, Instance(Scroll), help="""
+    Specify a scroll/pinch tool to be active when the plot is displayed.
+    """)
+
+    active_tap = Either(Auto, Instance(Tap), help="""
+    Specify a tap/click tool to be active when the plot is displayed.
+    """)
 
 class ToolbarBox(LayoutDOM):
     """ A layoutable toolbar that can accept the tools of multiple plots, and
@@ -103,7 +132,7 @@ class ToolbarBox(LayoutDOM):
     """)
 
 
-class PanTool(Tool):
+class PanTool(Drag):
     """ *toolbar icon*: |pan_icon|
 
     The pan tool allows the user to pan a Plot by left-dragging
@@ -128,7 +157,7 @@ class PanTool(Tool):
     """)
 
 
-class WheelZoomTool(Tool):
+class WheelZoomTool(Scroll):
     """ *toolbar icon*: |wheel_zoom_icon|
 
     The wheel zoom tool will zoom the plot in and out, centered on the
@@ -152,7 +181,7 @@ class WheelZoomTool(Tool):
     """)
 
 
-class SaveTool(Tool):
+class SaveTool(Action):
     """ *toolbar icon*: |save_icon|
 
     The save tool is an action. When activated, the tool opens a download dialog
@@ -168,7 +197,7 @@ class SaveTool(Tool):
     """
 
 
-class ResetTool(Tool):
+class ResetTool(Action):
     """ *toolbar icon*: |reset_icon|
 
     The reset tool is an action. When activated in the toolbar, the tool
@@ -187,7 +216,7 @@ class ResetTool(Tool):
     """)
 
 
-class ResizeTool(Tool):
+class ResizeTool(Drag):
     """ *toolbar icon*: |resize_icon|
 
     The resize tool allows the user to left-drag a mouse or drag a finger
@@ -199,7 +228,7 @@ class ResizeTool(Tool):
     """
 
 
-class TapTool(Tool):
+class TapTool(Tap):
     """ *toolbar icon*: |tap_select_icon|
 
     The tap selection tool allows the user to select at single points by
@@ -241,16 +270,10 @@ class TapTool(Tool):
     a dialog box, etc. See :class:`~bokeh.models.actions.Action` for details.
     """)
 
-@abstract
-class InspectTool(Tool):
-    pass
 
-    #active = Bool(True, help="""
-    #Whether the tool is intially active or not. If set to ``False``, the user
-    #will have to click tool's button to active it.
-    #""")
 
-class CrosshairTool(InspectTool):
+
+class CrosshairTool(Inspection):
     """ *toolbar icon*: |inspector_icon|
 
     The crosshair tool is a passive inspector tool. It is generally on
@@ -315,7 +338,7 @@ DEFAULT_BOX_OVERLAY = lambda: BoxAnnotation(
     line_dash=[4, 4]
 )
 
-class BoxZoomTool(Tool):
+class BoxZoomTool(Drag):
     """ *toolbar icon*: |box_zoom_icon|
 
     The box zoom tool allows users to define a rectangular
@@ -353,7 +376,7 @@ class BoxZoomTool(Tool):
     """)
 
 
-class BoxSelectTool(Tool):
+class BoxSelectTool(Drag):
     """ *toolbar icon*: |box_select_icon|
 
     The box selection tool allows users to make selections on a
@@ -419,7 +442,7 @@ DEFAULT_POLY_OVERLAY = lambda: PolyAnnotation(
     line_dash=[4, 4]
 )
 
-class LassoSelectTool(Tool):
+class LassoSelectTool(Drag):
     """ *toolbar icon*: |lasso_select_icon|
 
     The lasso selection tool allows users to make selections on a
@@ -468,7 +491,7 @@ class LassoSelectTool(Tool):
     """)
 
 
-class PolySelectTool(Tool):
+class PolySelectTool(Tap):
     """ *toolbar icon*: |poly_select_icon|
 
     The polygon selection tool allows users to make selections on a
@@ -504,7 +527,7 @@ class PolySelectTool(Tool):
     A shaded annotation drawn to indicate the selection region.
     """)
 
-class HoverTool(InspectTool):
+class HoverTool(Inspection):
     """ *toolbar icon*: |inspector_icon|
 
     The hover tool is a passive inspector tool. It is generally on at
@@ -636,7 +659,7 @@ class HoverTool(InspectTool):
 DEFAULT_HELP_TIP = "Click the question mark to learn more about Bokeh plot tools."
 DEFAULT_HELP_URL = "http://bokeh.pydata.org/en/latest/docs/user_guide/tools.html"
 
-class HelpTool(Tool):
+class HelpTool(Action):
     """
     The help tool is a widget designed to replace the hardcoded 'Help' link.
     The hover text can be customized through the ``help_tooltip`` attribute
@@ -651,7 +674,7 @@ class HelpTool(Tool):
     Site to be redirected through upon click.
     """)
 
-class UndoTool(Tool):
+class UndoTool(Action):
     """ *toolbar icon*: |undo_icon|
 
     Undo tool allows to restore previous state of the plot.
@@ -660,7 +683,7 @@ class UndoTool(Tool):
         :height: 18pt
     """
 
-class RedoTool(Tool):
+class RedoTool(Action):
     """ *toolbar icon*: |redo_icon|
 
     Redo tool reverses the last action performed by undo tool.

--- a/bokeh/plotting/figure.py
+++ b/bokeh/plotting/figure.py
@@ -6,7 +6,7 @@ logger = logging.getLogger(__name__)
 from ..io import curdoc, curstate
 from ..models import Plot
 from ..models import glyphs, markers
-from .helpers import _get_range, _process_axis_and_grid, _process_tools_arg, _glyph_function
+from .helpers import _get_range, _process_axis_and_grid, _process_tools_arg, _glyph_function, _process_active_tools
 from ..util._plot_arg_helpers import _convert_responsive
 
 DEFAULT_TOOLS = "pan,wheel_zoom,box_zoom,save,reset,help"
@@ -41,6 +41,10 @@ class Figure(Plot):
 
         title_text = kw.pop("title", None)
 
+        active_drag = kw.pop('active_drag', 'auto')
+        active_scroll = kw.pop('active_scroll', 'auto')
+        active_tap = kw.pop('active_tap', 'auto')
+
         super(Figure, self).__init__(*arg, **kw)
 
         self.title.text = title_text
@@ -51,8 +55,9 @@ class Figure(Plot):
         _process_axis_and_grid(self, x_axis_type, x_axis_location, x_minor_ticks, x_axis_label, self.x_range, 0)
         _process_axis_and_grid(self, y_axis_type, y_axis_location, y_minor_ticks, y_axis_label, self.y_range, 1)
 
-        tool_objs = _process_tools_arg(self, tools)
+        tool_objs, tool_map = _process_tools_arg(self, tools)
         self.add_tools(*tool_objs)
+        _process_active_tools(self.toolbar, tool_map, active_drag, active_scroll, active_tap)
 
     annular_wedge = _glyph_function(glyphs.AnnularWedge)
 

--- a/bokeh/plotting/helpers.py
+++ b/bokeh/plotting/helpers.py
@@ -287,9 +287,10 @@ def _process_tools_arg(plot, tools):
             function. I.e.: `wheel_zoom,box_zoom,reset`.
 
     Returns:
-        list of Tools objects added to plot
+        list of Tools objects added to plot, map of supplied string names to tools
     """
     tool_objs = []
+    tool_map = {}
     temp_tool_str = ""
     repeated_tools = []
 
@@ -310,6 +311,7 @@ def _process_tools_arg(plot, tools):
 
         tool_obj = _tool_from_string(tool)
         tool_objs.append(tool_obj)
+        tool_map[tool] = tool_obj
 
     for typename, group in itertools.groupby(
             sorted([tool.__class__.__name__ for tool in tool_objs])):
@@ -319,7 +321,46 @@ def _process_tools_arg(plot, tools):
     if repeated_tools:
         warnings.warn("%s are being repeated" % ",".join(repeated_tools))
 
-    return tool_objs
+    return tool_objs, tool_map
+
+
+def _process_active_tools(toolbar, tool_map, active_drag, active_scroll, active_tap):
+    """ Adds tools to the plot object
+
+    Args:
+        toolbar (Toolbar): instance of a Toolbar object
+        tools_map (dict[str]|Tool): tool_map from _process_tools_arg
+        active_drag (str or Tool): the tool to set active for drag
+        active_scroll (str or Tool): the tool to set active for scroll
+        active_tap (str or Tool): the tool to set active for tap
+
+    Returns:
+        None
+
+    Note:
+        This function sets properties on Toolbar
+    """
+    if active_drag in ['auto', None] or isinstance(active_drag, Tool):
+        toolbar.active_drag = active_drag
+    elif active_drag in tool_map:
+        toolbar.active_drag = tool_map[active_drag]
+    else:
+        raise ValueError("Got unknown %r for 'active_drag', which was not a string supplied in 'tools' argument" % active_drag)
+
+    if active_scroll in ['auto', None] or isinstance(active_scroll, Tool):
+        toolbar.active_scroll = active_scroll
+    elif active_scroll in tool_map:
+        toolbar.active_scroll = tool_map[active_scroll]
+    else:
+        raise ValueError("Got unknown %r for 'active_scroll', which was not a string supplied in 'tools' argument" % active_scroll)
+
+    if active_tap in ['auto', None] or isinstance(active_tap, Tool):
+        toolbar.active_tap = active_tap
+    elif active_tap in tool_map:
+        toolbar.active_tap = tool_map[active_tap]
+    else:
+        raise ValueError("Got unknown %r for 'active_tap', which was not a string supplied in 'tools' argument" % active_tap)
+
 
 _arg_template = "    %s (%s) : %s (default %r)"
 _doc_template = """ Configure and add %s glyphs to this Figure.

--- a/bokehjs/src/coffee/models/tools/toolbar.coffee
+++ b/bokehjs/src/coffee/models/tools/toolbar.coffee
@@ -1,5 +1,7 @@
 _ = require "underscore"
 
+p = require "../../core/properties"
+
 ActionTool = require "./actions/action_tool"
 HelpTool = require "./actions/help_tool"
 GestureTool = require "./gestures/gesture_tool"
@@ -45,9 +47,33 @@ class Toolbar extends ToolbarBase.Model
       if tools.length == 0
         continue
       @gestures[et].tools = _.sortBy(tools, (tool) -> tool.default_order)
-      if et not in ['pinch', 'scroll']
-        @gestures[et].tools[0].set('active', true)
 
+      if et == 'tap'
+        if @active_tap is null
+          continue
+        if @active_tap is 'auto'
+          @gestures[et].tools[0].active = true
+        else
+          @active_tap.active = true
+
+      if et == 'pan'
+        if @active_drag is null
+          continue
+        if @active_drag is 'auto'
+          @gestures[et].tools[0].active = true
+        else
+          @active_drag.active = true
+
+      if et in ['pinch', 'scroll']
+        if @active_scroll is null or @active_scroll is 'auto'
+          continue
+        @active_scroll.active = true
+
+  @define {
+      active_drag:   [ p.Any, 'auto' ]
+      active_scroll: [ p.Any, 'auto' ]
+      active_tap:    [ p.Any, 'auto' ]
+  }
 
 module.exports =
   Model: Toolbar

--- a/bokehjs/src/coffee/models/tools/toolbar_base.coffee
+++ b/bokehjs/src/coffee/models/tools/toolbar_base.coffee
@@ -99,12 +99,12 @@ class ToolbarBase extends LayoutDOM.Model
     return constraints
 
   @define {
-      tools:             [ p.Array,    []                     ]
-      logo:              [ p.String,   'normal'               ] # TODO (bev)
+      tools: [ p.Array,    []       ]
+      logo:  [ p.String,   'normal' ] # TODO (bev)
   }
 
   @internal {
-    gestures:   [ p.Any, () -> {
+    gestures: [ p.Any, () -> {
       pan:       { tools: [], active: null }
       tap:       { tools: [], active: null }
       doubletap: { tools: [], active: null }

--- a/sphinx/source/docs/user_guide/tools.rst
+++ b/sphinx/source/docs/user_guide/tools.rst
@@ -52,6 +52,18 @@ running the code and changing the ``toolbar_location`` value.
 Specifying Tools
 ----------------
 
+At the lowest |bokeh.models| level, tools are added to a ``Plot`` by
+passing instances of ``Tool`` objects to the ``add_tools`` method:
+
+.. code-block:: python
+
+    plot = Plot()
+    plot.add_tools(LassoSelectTool())
+    plot.add_tools(WheelZoomTool())
+
+This explicit way of adding tools works with any Bokeh ``Plot`` or
+``Plot`` subclass, such as ``Figure`` or ``Chart``.
+
 Tools can be specified by passing the ``tools`` parameter to the |figure|
 function or to any |bokeh.charts| Chart function. The tools parameter
 accepts a list of tool objects, for instance:
@@ -65,7 +77,7 @@ containing tool shortcut names:
 
 .. code-block:: python
 
-    tools = "pan,wheel_zoom,box_zoom,reset,resize"
+    tools = "pan,wheel_zoom,box_zoom,reset"
 
 However, this method does not allow setting properties of the tools.
 To use shortcut names but also add tools with properties, one can
@@ -73,8 +85,52 @@ also call the ``add_tools`` method:
 
 .. code-block:: python
 
-    fig = figure(tools="pan,wheel_zoom,box_zoom,reset,resize")
-    fig.add_tools(BoxSelectTool(dimensions=["width"]))
+    plot = figure(tools="pan,wheel_zoom,box_zoom,reset")
+    plot.add_tools(BoxSelectTool(dimensions=["width"]))
+
+.. _userguide_tools_setting_active_tools:
+
+Setting the Active Tools:
+-------------------------
+
+Bokeh toolbars can have (at most) one active tool from each kind of gesture
+(drag, scroll, tap). By default, Bokeh will use a default pre-defined
+order of preference to choose one of each kind from the set of configured
+tools, to be active.
+
+However it is possible to exert control over which tool is active. At the
+lowest |bokeh.models| level, this is accomplished by using the ``active_drag``,
+``active_scroll``, and ``active_tap`` properties of ``Toolbar``. These
+properties can take the following values:
+
+* ``None`` --- there is no active tool of this kind
+* ``"auto"`` --- Bokeh chooses a tool of this kind to be active (possibly none)
+* a ``Tool`` instance --- Bokeh sets the given tool to be the active tool
+
+As an example:
+
+.. code-block:: python
+
+    # configure so that no drag tools are active
+    plot.toolbar.active_drag = None
+
+    # configure so that Bokeh chooses what (if any) scroll tool is active
+    plot.toolbar.active_scroll = "auto"
+
+    # configure so that a specific PolySelect tap tool is active
+    plot.toolbar.active_tap = poly_select
+
+The default value for all of these properties is ``"auto"``.
+
+Active tools can be specified by passing the these properties as keyword
+arguments to the |figure| function or to any |bokeh.charts| Chart function.
+In this case, it is also possible to pass any one of the string names for,
+ease of configuration:
+
+.. code-block:: python
+
+    # configures the lasso tool to be active
+    plot = figure(tools="pan,lasso_select,box_select", active_drag="lasso_select")
 
 .. _userguide_tools_pandrag:
 

--- a/sphinx/source/docs/user_guide/tools.rst
+++ b/sphinx/source/docs/user_guide/tools.rst
@@ -52,7 +52,7 @@ running the code and changing the ``toolbar_location`` value.
 Specifying Tools
 ----------------
 
-At the lowest |bokeh.models| level, tools are added to a ``Plot`` by
+At the lowest ``bokeh.models`` level, tools are added to a ``Plot`` by
 passing instances of ``Tool`` objects to the ``add_tools`` method:
 
 .. code-block:: python
@@ -90,8 +90,8 @@ also call the ``add_tools`` method:
 
 .. _userguide_tools_setting_active_tools:
 
-Setting the Active Tools:
--------------------------
+Setting the Active Tools
+------------------------
 
 Bokeh toolbars can have (at most) one active tool from each kind of gesture
 (drag, scroll, tap). By default, Bokeh will use a default pre-defined
@@ -99,7 +99,7 @@ order of preference to choose one of each kind from the set of configured
 tools, to be active.
 
 However it is possible to exert control over which tool is active. At the
-lowest |bokeh.models| level, this is accomplished by using the ``active_drag``,
+lowest ``bokeh.models`` level, this is accomplished by using the ``active_drag``,
 ``active_scroll``, and ``active_tap`` properties of ``Toolbar``. These
 properties can take the following values:
 
@@ -131,6 +131,11 @@ ease of configuration:
 
     # configures the lasso tool to be active
     plot = figure(tools="pan,lasso_select,box_select", active_drag="lasso_select")
+
+.. _userguide_tools_builtin_tools:
+
+Built-in Tools
+--------------
 
 .. _userguide_tools_pandrag:
 

